Tests/Microsoft.AppCenter.Test.Windows/Microsoft.AppCenter.Test.Windows.csproj

```diff
@@ -8,7 +8,7 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>Microsoft.AppCenter.Test.Windows</RootNamespace>
     <AssemblyName>Microsoft.AppCenter.Test.Windows</AssemblyName>
-    <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.8</TargetFrameworkVersion>
```

Comment:
Currently, the minimum supported version is 4.6.1: https://docs.microsoft.com/en-us/appcenter/sdk/getting-started/wpf-winforms#1-prerequisites
What requires the version bump?

Reply:
greeting you can't test for SecurityProtocolType.SystemDefault if the test targets prior 4.7 - i'd only done the bump for the test not the sdk itself

Reply:
Hi! Thanks for your contribution PR!
I checked these changes locally and unfortunately it doesn't work for .NET Framework projects.

The error which I got during testing:
```
2021-06-04 13:34:35.483 [AppCenter] ERROR: Sending logs for channel 'core', batch 'f221ebf6-9fe7-4c50-b529-53196bad0224' failed: System.PlatformNotSupportedException: Operation is not supported on this platform.
at System.Net.Http.HttpClientHandler.set_SslProtocols(SslProtocols value)
at Microsoft.AppCenter.Ingestion.Http.HttpNetworkAdapter.<>c.<.cctor>b__0_0()
at Microsoft.AppCenter.Ingestion.Http.HttpNetworkAdapter.get_HttpClient()
at Microsoft.AppCenter.Ingestion.Http.HttpNetworkAdapter.<SendRequestAsync>d__12.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
at Microsoft.AppCenter.Ingestion.Http.HttpNetworkAdapter.<SendAsync>d__10.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
at Microsoft.AppCenter.Ingestion.Http.IngestionHttp.<CallAsync>d__9.MoveNext()
```

This happens because this `HttpClientHandler.SslProtocols` property available only since .NET Framework version 4.7.1 (see [here](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclienthandler.sslprotocols?view=net-5.0#applies-to)).
But minimal .NET Framework [version of App Center SDK](https://docs.microsoft.com/en-us/appcenter/sdk/#wpf-and-winforms) is 4.6.1.

Also, I was a bit confused by [these changes](https://github.com/microsoft/appcenter-sdk-dotnet/pull/1532/files#diff-599600bcb331f8e91a16fec61aea0d947a209eaa14bfce51668a0b196a24e8e4R11 ) because all changes should work (including tests) with the minimal version.

This issue looks like a feature request. We will create a ticket about it and take care of it ourselves.

Reply:
This will be the fun with the behaviour change done in NET47 :)
Currently using the AppCenter SDK on NET47 or later breaks connections to any site not using TLS1.2 (i.e. 1.1 or 1.3) i appreciate TLS1.1 is supposed to be phased out.
One option is to change the SDK to say `if ServicePointManager.SecurityProtocol = 0` (because NET461 doesn't know about `SystemDefault` so can't use the enum itself) then don't do `ServicePointManager.SecurityProtocol |= Tls12` your challenge there is if the OS has TLS1.2 disabled on SystemDefault but that's a weird edge case in that circumstance?
Other options are
* Multi target the SDK (so NET47 onward can use the newer httpclient, NET46 does the !=0 check) i didn't want to litter your codebase with #if NET461 - it's not my place to do that :)
* Put it as a known issue somewhere in the documentation and explain people have to set `ServicePointManager.SecurityProtocol |= SslProtocol.TLS13`  (etc) directly after AppCenter.Start() if they using NET47 or later and see connection issues.
* Remove changing the SslProtocol completely and have it documented that people need to ensure they have TLS1.2 enabled

As I said happy to help out, it's one of those niggly ones. I do have a workaround as I know I can turn the desired protocol(s) back on. But it might confuse others if their sites etc randomly stop working upon updating their framework from NET462 to NET47 (if still using tls1.1) or NET5.

the simplest change right now (and most sensible?) is something along the lines of

```
if (ServicePointManager.SecurityProtocol != 0)
{
    ServicePointManager.SecurityProtocol |= ServicePointManager.Tls12
}
```

this means NET461 continues as is, and later platforms use the OS default or add to anything the developer has already done to set specific protocols. The change to HttpClient instead of ServicePointManager is a side issue for the longer term :)

Hope this helps
Many Thanks

Reply:
Thanks for your detailed answer. We have already created a feature request about it. We will notify you when we will have any updates.